harvester/harvester
# fix(backup): skip vmimage restore if there is same display name

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
If there is same display name VM image in the cluster, the restoring metadata from backup target fails. The reason is that VM image can't be created and the whole process is stuck.

**Solution:**
Skip restoring VM image if there is another VM image which has same display name in the cluster.

**Related Issue:**
https://github.com/harvester/harvester/issues/6772

**Test plan:**
Case 1: Removing VM image can cleanup VM image metadata in backup target

1. Create a new cluster (this PR updates CRD, so please create a new cluster from this PR)
2. Setup backup target.
3. Create a VM image and VM.
4. Take a backup for the VM.
5. Check there is VM image metadata in the backup target.
6. Remove the backup, VM, and VM image.
7. Check the VM image metadata is removed in the backup target.

Case 2: Skip restoring VM image with same display name

1. Create a new cluster (this PR updates CRD, so please create a new cluster from this PR)
2. Setup backup target.
3. Create a VM image and VM.
4. Take a backup for the VM.
5. Set backup target to default value.
6. Remove the backup, VM, and VM image.
7. Create a new VM image with same display name and different CR name.
8. Set backup target back.
9. Check harvester pod has following log.
```
skip create vm image, because there is already an image with the same display name

skip creating vm backup, because the storage class is not found
```
